java/aiservices/openai/src/main/java/com/microsoft/semantickernel/aiservices/openai/chatcompletion/OpenAIChatCompletion.java

```diff
@@ -369,33 +332,67 @@ private static ChatCompletionsOptions getCompletionsOptions(
-        @Nullable
-        List<FunctionDefinition> functions) {
+    private static void configureToolCallBehaviorOptions(
+            ChatCompletionsOptions options,
+            @Nullable
+            ToolCallBehavior toolCallBehavior,
+            @Nullable
+            List<OpenAIFunction> functions,
+            int autoInvokeAttempts) {
 
         if (functions == null || functions.isEmpty()) {
-            return Collections.emptyList();
+            return;
         }
 
-        if (toolCallBehavior == null || !(toolCallBehavior.kernelFunctionsEnabled()
-            || toolCallBehavior.autoInvokeEnabled())) {
-            // If tool calls are not explicitly enabled, then we don't need to send any tool definitions
-            return Collections.emptyList();
+        if (toolCallBehavior == null || autoInvokeAttempts == 0) {
+            // if auto-invoked is not enabled, then we don't need to send any tool definitions
+            return;
+        }
+
+        // If a specific function is required to be called
+        KernelFunction<?> toolChoice = toolCallBehavior.functionRequired();
+        if (toolChoice != null) {
+            List<ChatCompletionsToolDefinition> toolDefinitions = new ArrayList<>();
+            toolDefinitions.add(new ChatCompletionsFunctionToolDefinition(
+                    OpenAIFunction.toFunctionDefinition(
+                        toolCallBehavior.functionRequired().getMetadata(),
+                        toolCallBehavior.functionRequired().getPluginName()
+                    ))
+            );
+
+            options.setTools(toolDefinitions);
+            try {
+                String json = String.format("{\"type\":\"function\",\"function\":{\"name\":\"%s%s%s\"}}",
+                        toolChoice.getPluginName(),
+                        OpenAIFunction.getNameSeparator(),
+                        toolChoice.getName()
+                );
+                options.setToolChoice(BinaryData.fromObject(new ObjectMapper().readTree(json)));
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+            return;
         }
 
-        return functions.stream()
+        List<ChatCompletionsToolDefinition> toolDefinitions = functions.stream()
             .filter(function -> {
-                String[] parts = function.getName().split("-");
-                String pluginName = parts.length > 0 ? parts[0] : "";
-                String fnName = parts.length > 1 ? parts[1] : "";
-                return toolCallBehavior.functionEnabled(pluginName, fnName);
+                // if kernel functions are enabled we send all tool definitions
+                if (toolCallBehavior.kernelFunctionsEnabled()) {
+                    return true;
+                }
+                // otherwise, check if the function is enabled
+                return toolCallBehavior.functionEnabled(function.getPluginName(), function.getName());
```

Comment:
Does ToolCallBehavior#functionEnabled return true if kernelFunctionsEnabled and the function is not explicitly disabled? I think returning true on line 381 might be a bug if the kernel function was explicitly disabled.